mypy/checkexpr.py

```diff
@@ -2774,38 +2774,34 @@ def check_boolean_op(self, e: OpExpr, context: Context) -> Type:
 
         assert e.op in ('and', 'or')  # Checked by visit_op_expr
 
-        if e.op == 'and':
+        left_map: mypy.checker.TypeMap
+        right_map: mypy.checker.TypeMap
+        if e.right_always:
```

Comment:
For always-true/false cases, `find_isinstance_check` returns a `{}, None` or a `None, {}` pair. We do the same here, but without even calling `find_isinstance_check`.